File: modules/angular2/test/web_workers/shared/mock_event_emitter.dart
library angular2.test.web_workers.worker.mock_event_emitter;

import 'dart:core';
import 'dart:async';
import "package:angular2/src/facade/async.dart";

class MockEventEmitter<T> extends EventEmitter<T> {
  final controller = new StreamController.broadcast(sync: true);

  @override
  StreamSubscription listen(void onData(dynamic line),
      {void onError(Error error), void onDone(), bool cancelOnError}) {
    return controller.stream.listen(onData,
        onError: onError, onDone: onDone, cancelOnError: cancelOnError);
  }

  @override
  void add(value) {
    controller.add(value);
  }
}

File: modules/angular2/test/web_workers/shared/mock_event_emitter.ts
import {EventEmitter} from 'angular2/src/core/facade/async';

export class MockEventEmitter<T> extends EventEmitter<T> {
  private _nextFns: Function[] = [];

  constructor() { super(); }

  subscribe(next: any): any {
    this._nextFns.push(next);
    return new MockDisposable();
  }

  next(value: any) { this._nextFns.forEach(fn => fn(value)); }
}

class MockDisposable {
  isUnsubscribed: boolean = false;
  unsubscribe(): void {}
}
